src/riak_kv_qry_compiler.erl

```diff
@@ -172,6 +171,19 @@ expand_query(?DDL{table = Table, local_key = LK, partition_key = PK},
             {ok, SubQueries2}
     end.
 
+%% Check if any columns in the partition key have the DESC keyword applied.
+%%
+%% Only check the last column in the partition key, otherwise the start/end key
+%% does not need to be flipped. The data is not stored in a different order to
+%% the start/end key.
+%%
+%% Checking the last column in the partition key is safe while the other columns
+%% must be exactly specified e.g. anything but the QUANTUM column must be
+%% covered with an equals operator in the where clause.
+is_partition_key_cols_descending(FieldOrders, #key_v1{ast = AST}) ->
+    PKOrders = lists:sublist(FieldOrders, length(AST)),
```

Comment:
Untested, possibly buggy. I could live with embedding `length(AST)` with a comment as well, I just like this approach a bit better.

```
is_partition_key_cols_descending(FieldOrders, PK) ->
   lists:nth(num_cols(PK)) == descending.

%% The abstract syntax tree for the key record contains one element per columns
num_cols(#key_v1{ast = AST}) ->
    length(AST).
```